### ISSUE TITLE
French Translation infrastructure pages

### ISSUE THREAD

**Issue by acrymble** [programminghistorian/jekyll#836]
Before we can publish any French lessons, we need the following pages translated, reviewed, and published:

- [x] index.md 
- [x] README.md
- [x] 404.md
en/
- [x] about.md
- [x] author-guidelines.md
- [x] contribute.md
- [x] editor-guidelines.md
- [x] feedback.md
- [x] lesson-retirement-policy.md
- [x] lessons.md
- [x] project-team.md
- [x] research.md
- [x] reviewer-guidelines.md
- [x] /_data/activities.yml
- [x] /_data/snippets.yml
- [x] /_data/teamroles.yml
- [x] /_data/topics.yml
- [x] /_data/ph_authors.yml ("team: true" members only)

---
To be updated by website team:

- [ ] /_plugins/validate_yaml.rb

**Comment by spapastamkou** [comment]
Should we include as well a translation of the "Technical Contributions" part of the wiki as discussed in #847 ?

**Comment by drjwbaker** [comment]
@arojascastro Do we have a prior justification for not translating this section?

**Comment by arojascastro** [comment]
I do not think so @drjwbaker. If we are moving that section from the wiki to the main page, we should translate it. @jenniferisasi @vgayolrs @mariajoafana @jamotilla anyone wants to try it?

**Comment by acrymble** [comment]
potential for frequent change was the only reason. So if everyone wants it
translated, that's fine. It isn't really needed for the general public.

**Comment by drjwbaker** [comment]
Okay. That makes sense. If frequent change is the issue, maybe it can be decided by each language initiative if they need their own Technical Contributions page? (eg, there may be language specific materials out there that cover what is required) I'm imagining that as each language initiative diverges per https://github.com/programminghistorian/jekyll/wiki/Additional-Language-Sub-Teams-Policy they may each need their own wiki materials to plan/execute their work.

**Comment by mdlincoln** [comment]
Right - because there are frequent changes AND because it is about the internal operations of the journal, and so used by a far smaller number of people, I don't believe it is worth the considerable effort involved to publish them.

**Comment by mdlincoln** [comment]
I will set up a new branch for `french-infrastructure` and start a `/fr` directory for French team members to begin editing.

**Comment by spapastamkou** [comment]
Thank you. 
I suppose we should update and start use the Reviewer Tracking googlesheet that is available at #600 , is that correct?

**Comment by amsichani** [comment]
I think we're still waiting for @mdlincoln to set up the new branch and fr/ directory - do you want me to set them up ?
then, @spapastamkou and @mpuren we should start translating pages - it would be good to allocate things and set up a workflow alongside a timeframe. I am happy to help with the overall procedure while refreshing my French. 
@spapastamkou I think the reviewer tracking gsheet is open for update, if you feel ready to start adding reviewers for lessons - if I may, I 'd suggest to finish first with the fr/ infra (there is a lot to do) and then move to the lessons/reviews bit. how that sounds?

**Comment by spapastamkou** [comment]
thank you @amsichani. Overall, I finish some pressing obligations in the days to come and I can be regularly available for this work to progress, roughly from the week of 25 June until the end of July. 

For the 1st point: Let's see how @mdlincoln prefers we proceed. If it is up to the new members to do it, we will. In any case I am okay with whatever solution is judged best for the team (and does not charge others with extra work). 

For the 2nd: I propose [this sheet](https://framacalc.org/ph-fr-infra-pages) as a base to start organize ourselves. We can change/adapt if needed, please feel free to comment. I provided for two reviews following the pattern for the lessons but I would appreciate confirmation on that. I also propose that somewhere around mid-July we check how things advance and readjust if necessary. What do you think @amsichani et @mpuren? Are you available?  

For the 3rd: I mentionned the gsheet only bcz I thought possible to use it also for the infra pages but apparently it is not the case:) You are right, let's leave that for when we are at the next stage.

**Comment by acrymble** [comment]
@amandavisconti @spapastamkou there is already a /fr directory on the submissions site. Everything should be put there until it is ready to publish. https://github.com/programminghistorian/ph-submissions/tree/gh-pages/fr

You don't need external peer reviewers for the infrastructure pages (I don't think), as long as both you and @mpuren are happy with the text. Unless you think it is worth doing.

**Comment by amsichani** [comment]
I saw this one @acrymble but I thought we were planning to have a new branch and a fr/ directory on the jekyll site for the infrastructure pages (as we 've done with the es/)
@spapastamkou I am fine with that plan - mid July sounds good for a recap. I will keep an eye on the progress, offer review and help, if needed - just drop me a line.

**Comment by acrymble** [comment]
If you put them on the jekyll site people can see them. We don't want that until there's some content for them to look at.

**Comment by amsichani** [comment]
okey, I thought a new branch on the Jekyll site was just enough for this type of experimentations  ;-)

**Comment by mdlincoln** [comment]
And to clarify, I will create a new branch on THIS repository so that @programminghistorian/french-team can begin translating all of the other _non-lesson pages_ there, while the lesson translations go through the usual process on the submissions repo. I'll post here when that's ready - sorry, my schedule had been overtaken by events at work this past week.

**Comment by mdlincoln** [comment]
@programminghistorian/french-team I've now added a new branch to this repository: [`french-infrastructure`](https://github.com/programminghistorian/jekyll/tree/french-infrastructre) If you switch to that branch, you'll see a new `fr` directory with the content pages - I just copied over the ones from `en`.

Youu can see a preview version of this branch at https://french-infrastructre--ph-dev.netlify.com/fr 

Consult that page to see how your translation looks online. It usually takes Netlify a few minutes to build the site after new commits are pushed to the branch, so please be patient.

First step: Please rename all the files in `fr/` so that they have French names EXCEPT `index.md` - that file name must stay the same. The name you give the file will determine the URL. For example, if you rename `lessons.md` to `cours.md`, then it will eventually be published at the URL `https://programminghistorian.org/fr/cours`. Also, I am sorry to say that you must only use ASCII characters in the file names, so no diacritics are allowed because of an old issue with git and UTF-8 filenames 😢 **When renaming the files, please don't change the contents yet.** Commit and push all the changes and let me make sure everything looks ok, and then we can continue.

Second step: It is probably most important to translate `_data/snippets.yml` first. Without that, you'll see a lot of blank buttons and menus all over the site, and so it will be hard to navigate the preview site. The `snippets.yml` file contains small fragments of text that show up on all pages but need to appear in multiple languages. Generally, the structure looks like this:

https://github.com/programminghistorian/jekyll/blob/e233575eaca5ec230631cfcf31f308edea911103/_data/snippets.yml#L364-L367

I have tried to add empty slots for `fr:` to all the necessary parts of the file. One special section is at the top of `snippets.yml` where the different menu labels and links are defined, e.g.

https://github.com/programminghistorian/jekyll/blob/e233575eaca5ec230631cfcf31f308edea911103/_data/snippets.yml#L15-L24

There, you will need to translate the `title` and also insert the `link`, which should correspond to the file names that you selected in step one.

After that's done, you can start translating the contents of each page. Please post in this issue as you progress and I will be here for any questions you have!

**Comment by mdlincoln** [comment]
FYI you will occasionally see a special comment in the content of these files, like this:

```
{% comment %}
This creates a variable you can call to just pull lessons (which are the only pages with 'Lesson: true' in their front matter, courtesy of config.yml setting the scope of site URLs containing the 'lessons' path to follow 'lesson: true').
{% endcomment %}
```

You do not need to translate this text.

**Comment by arojascastro** [comment]
Sorry to jump into this, but maybe that information name about files and ASCII should be added to our guidelines: https://programminghistorian.org/en/author-guidelines#choose-a-searchable-name

I'm saying this because I recently chose a file name with an accent and it was displayed weirdly on the browser.

**Comment by spapastamkou** [comment]
a huge thanks @mdlincoln for all this. It is all very clear, @mpuren and I will keep you updated here. I will start working on all that around June 25 and regularly afterwards.
Thank you @acrymble and @amsichani as well. We'll make it simple for the reviews of the infra pages:) I count on your help @amsichani and will let you know in time:-)
I do not use diacritics in file names but @arojascastro has a good point there, it would be useful to insert it in the guidelines, don't you think?
Again, thank you all!

**Comment by acrymble** [comment]
Do we have any progress to report here? Some of the pages are very short and could be a good place to start to make sure you know how everything works. For example, 404.md only has 2 sentences to translate:

> We are sorry, but we can't find any content at URL you tried to visit.
> 
> Please contact us if you think there ought to be something here!

**Comment by amsichani** [comment]
We are having a series of Skype calls and intense working sessions next week with the FR task force so hopefully we ll be having few things to report on this during our regular Skype meeting.

**Comment by spapastamkou** [comment]
Hi to all. I have been working on the fr-infrastructure branch as agreed
with Matthew. The fr directory has been prepared <
https://github.com/programminghistorian/jekyll/commit/1973685eb99e0a465577ada479e02c416079ce38>
and I added fr translations in  _data <
https://github.com/programminghistorian/jekyll/tree/french-infrastructre/_data>.
I currently work on the about page which is almost finished. I did not tick
any boxes in the list of this ticket because I prefer my translations to
receive a second lecture before, which is one the subjects to discuss on
Tuesday on a skype meeting with Marie and Anna-Maria. I planned to open a
ticket on issues on the fr translations (and thus close #864) but it is for
later in this weekend. For the time being I keep the framasheet updated
each time I finish a task.

**Comment by spapastamkou** [comment]
@mpuren : 404.md, apropos.md, index.md added 21/07/18

**Comment by mpuren** [comment]
@spapastamkou Thanks. I'm going to review the translations next week as it is mentionned in the speadsheet (one day will be enough, I think), and I'll begin to translate the remaining texts in English and hopefully finish this work next week too.

**Comment by spapastamkou** [comment]
FYI @mpuren recherche, politique-retrait-lecons, equipe-projet just added

**Comment by spapastamkou** [comment]
for info, files that correspond to author guideliens, feedback and lessons pages have been translated in FR thanks to @mpuren. Some more to come soon:)

**Comment by spapastamkou** [comment]
contribute page in FR added
editor guidelines in FR added (changes in anchors should be made after publication)

**Comment by amsichani** [comment]
@spapastamkou and @mpuren , a quick update here from our Skype call : once you ll finish with translating & editing the branch, we need to inform @mdlincoln so he ll be able to secure some time to work through the integration. Keep up the good work, ladies, and please drop me a line if you need any help!

**Comment by amsichani** [comment]
@spapastamkou @mpuren , as I missed last week's Skype, I see that you have now finish with translation/ editing and we are now at the point to ask @mdlincoln 's magic for merging the branch -he needs to secure extra time for this and it d be nice to have a more-or-less concrete timeframe for this. Do let me know where we are on this and if you need any help.

**Comment by mdlincoln** [comment]
Thank you for the huge amount of translation work you've all done @programminghistorian/french-team! Here's my plan for the next steps:

1. I'll do a first pass at bringing this branch up to date with all the changes that have happened in the last few months.
2. We will need to do a pass through both the snippets and the team bios, as some of these have been slightly changed and so there are a few parts that need to be re-translated for French pages.
3. Start loading some of the French lesson translations.
3. I'll open tickets for any technical bugs that will need fixing (e.g. I think there's still some issues I need to fix with the lesson directory)
4. We figure out a plan to do a systematic review of all the new pages prior to launch.

I'm going to close this issue now, and begin to open up some individual tasks - some for myself, and some for the @programminghistorian/french-team.
